### PR TITLE
Issue #6630: Stabilize ExceptionEventData

### PR DESCRIPTION
**Pull request by olamideTiana**
#6630 - Stabilize ExceptionEventData

The class is now well-structured for stability and immutability and It correctly handles input validation and provides a clean interface for accessing its data.
@jack-berg Please kindly review, I'll appreciate your feedback on it and let me know if there's any corrections I can make

### CODE REVIEW

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/data/ExceptionEventData.java [hunk 1]
```diff
@@ -9,43 +9,82 @@
...
-
-  /**
-   * Return the additional {@link Attributes attributes} of the {@link ExceptionEventData}.
-   *
-   * @return the additional {@link Attributes attributes} of the {@link ExceptionEventData}
-   */
-  Attributes getAdditionalAttributes();
+public final class ExceptionEventData {
+
+// Fields are private and final for immutability
```
Comment (jack-berg) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/data/ExceptionEventData.java:14]: We use a tool called spotless to ensure consistent code formatting. You'll need to run `./gradlew spotlessApply` and commit changes for the build to pass.

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/data/ExceptionEventData.java [hunk 2]
```diff
@@ -9,43 +9,82 @@
 import io.opentelemetry.sdk.trace.SpanLimits;
 import io.opentelemetry.sdk.trace.data.EventData;
 
-/**
- * Data representation of an event for a recorded exception.
```
Comment (jack-berg) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/data/ExceptionEventData.java:13]: Let's keep this comment.

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/data/ExceptionEventData.java
Comment (jack-berg) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/data/ExceptionEventData.java:1]: We'll need to move this to `io.opentelemetry.sdk.trace.data`. Internal packages are for internal APIs not meant for public consumption.

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/data/ExceptionEventData.java (hunk 1 above)
Comment (jack-berg) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/data/ExceptionEventData.java:14]:
```suggestion
```

**Review by jack-berg — COMMENTED** [review]
There are the variety of basic problems with this PR. I'd normally spend time addressing each problem individually, but this looks like the output of an LLM, and so I'm reluctant to spend time on it.

So for now I'll just leave some high level comments:

- Check out the [contributing](https://github.com/open-telemetry/opentelemetry-java/blob/main/CONTRIBUTING.md) guide
- Follow the example of other code in the repository
- I don't care if you use llms as a starting spot, but a raw copy / paste of the output is rarely going to work, and is an easy way to loose maintainer confidence. After all, why should I spend time / effort with a thoughtful review if the effort isn't reciprocated? Its easier (and more enjoyable) to do the work myself.

**Comment by olamideTiana** [comment]
@jack-berg My apologies for any inconveniences caused, I'll revert the changes made

**Pull request closed without merging**